infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Action.java

```diff
@@ -0,0 +1,25 @@
+package org.corfudb.infrastructure.orchestrator;
+
+import org.corfudb.runtime.CorfuRuntime;
+
+/**
+ *
+ * An idempotent workflow action. All workflow actions must extend this class.
+ *
+ * Created by Maithem on 10/25/17.
+ */
+
+public abstract class Action {
+
+    /**
+     * Returns the name of this action.
+     * @return Name of action
+     */
+    public abstract String getName();
+
+    /**
+     * Execute the action.
+     * @param runtime A that the action will use to execute
```

Comment:
If it needs to be idempotent, it better be in the javadoc that it needs to be.

Reply:
Changed the language for now.